golem-worker-service/src/gateway_api_definition/http/api_oas_convert.rs

```diff
@@ -0,0 +1,1285 @@
+    param_type: openapiv3::Schema,
+) -> openapiv3::Parameter {
+    let parameter_data = openapiv3::ParameterData {
+        name: param_name.to_string(),
+        description: Some(format!("Header parameter: {param_name}")),
+        required: true,
+        deprecated: None,
+        explode: Some(false),
+        format: openapiv3::ParameterSchemaOrContent::Schema(openapiv3::ReferenceOr::Item(
+            param_type,
+        )),
+        example: None,
+        examples: Default::default(),
+        extensions: Default::default(),
+    };
+
+    openapiv3::Parameter::Header {
+        parameter_data,
+        style: openapiv3::HeaderStyle::Simple,
+    }
+}
+
+// Helper function: Adds request body to the operation
+fn add_request_body(operation: &mut openapiv3::Operation, route: &CompiledRoute) {
+    // Only add request body if we have a binding that contains response_mapping
+    match &route.binding {
+        GatewayBindingCompiled::Worker(worker_binding) => {
+            if let Some(request_body) =
+                create_request_body(route, &worker_binding.response_compiled.rib_input)
+            {
+                operation.request_body = Some(openapiv3::ReferenceOr::Item(request_body));
+            }
+        }
+        GatewayBindingCompiled::FileServer(file_server_binding) => {
+            if let Some(request_body) =
+                create_request_body(route, &file_server_binding.response_compiled.rib_input)
+            {
+                operation.request_body = Some(openapiv3::ReferenceOr::Item(request_body));
+            }
+        }
+        _ => {}
+    }
+}
+
+// Helper function: Determines response schema
+fn determine_response_schema_and_content_type(
+    route: &CompiledRoute,
+) -> (Option<openapiv3::Schema>, String) {
+    if let GatewayBindingCompiled::Worker(worker_binding) = &route.binding {
+        if let Some(output_info) = &worker_binding.response_compiled.rib_output {
+            if let AnalysedType::Record(record) = &output_info.analysed_type {
+                let (headers_opt, body_opt, status_opt) = extract_response_fields(&record.fields);
+                let content_type = "application/json".to_string();
+                if headers_opt.is_some() || body_opt.is_some() || status_opt.is_some() {
+                    let schema = body_opt.map(|body| create_schema_from_analysed_type(&body));
+                    return (schema, content_type);
+                } else {
+                    // No structured response fields, use entire record
+                    let schema = Some(create_schema_from_analysed_type(&output_info.analysed_type));
+                    return (schema, content_type);
+                }
+            }
+            // Not a record, use type as-is
+            let schema = Some(create_schema_from_analysed_type(&output_info.analysed_type));
+            return (schema, "application/json".to_string());
+        }
+    }
+    (None, "application/json".to_string())
```

Comment:
I understand this logic is due to the complexity that we discussed previously. I am ok for this for now.